src/metadata/selectors.js

```diff
@@ -85,6 +85,29 @@ export const getCategoriesByCategoryComboId = createCachedSelector(
         categoryCombo.categories.map((id) => categories[id])
 )((_, categoryComboId) => categoryComboId)
 
+/**
+ * @param {*} metadata
+ * @param {*} dataSetId
+ */
+export const getCategoryComboByDataSetId = createCachedSelector(
+    getDataSetById,
+    getCategoryCombos,
+    (dataSet, categoryCombos) => {
+        return categoryCombos?.[dataSet.categoryCombo.id]
+    }
+)((_, categoryComboId) => categoryComboId)
+
+/**
+ * @param {*} metadata
+ * @param {*} dataSetId
+ */
+export const getCategoriesByDataSetId = createCachedSelector(
+    getCategoryComboByDataSetId,
+    getCategories,
+    (categoryCombo, categories) =>
```

Comment:
There is a selector for `getCategoriesByCategoryComboId`, so you could probably do something like

```javascript
export const getCategoriesByDataSetId = createCachedSelector(
     metadata => metadata, 
    getCategoryComboByDataSetId,
    (metadata, categoryCombo) => getCategoriesByCategoryComboId(metadata, categoryCombo.id)
```

Not sure if this is any cleaner though, since `getCategoryComboByDataSetId` - returns the object and not just the ID.

Also it doesn't look like that function is checking for the existence of the `categoryCombo` - which I think it should. 
So if you decide to use that, I think we should add the defensive coding there as well.